src/mame/mr/dribling.cpp

```diff
@@ -246,46 +269,56 @@ void dribling_state::misc_w(uint8_t data)
 		m_maincpu->set_input_line(0, CLEAR_LINE);
 
 	// bit 6 = parata (save by goalkeeper)
-
+	// TODO: this sound is very slow, needs to be fixed.
+	// m_i_parata->write_line(BIT(data, 6));
 	// bit 5 = ab. campo (field enable)
 	m_abca = (data >> 5) & 1;
-
 	// bit 4 = ab. a.b.f.
+	m_i_enable->write_line(BIT(data, 4));
+
+
+
+
 	// bit 3 = n/c
 
 	// bit 2 = (9) = PC2
 	// bit 1 = (10) = PC1
 	// bit 0 = (32) = PC0
 	m_input_mux = data & 7;
-	LOGMISC("%s:misc_w(%02X)\n", machine().describe_context(), data);
-}
 
+}
 
 void dribling_state::sound_w(uint8_t data)
 {
-	// bit 7 = stop palla (ball stop)
-	// bit 6 = contrasto (tackle)
-	// bit 5 = calcio a (kick a)
-	// bit 4 = fischio (whistle)
-	// bit 3 = calcio b (kick b)
-	// bit 2 = folla a (crowd a)
-	// bit 1 = folla m (crowd m)
-	// bit 0 = folla b (crowd b)
-	LOGSOUND("%s:sound_w(%02X)\n", machine().describe_context(), data);
+	m_i_stop_palla->write_line(BIT(data, 7));
+	m_i_contrasto->write_line(BIT(data, 6));
+	m_i_calcio_a->write_line(BIT(data, 5));
+	m_i_fischio->write_line(BIT(data, 4));
+	m_i_calcio_b->write_line(BIT(data, 3));
+	m_i_folla_a->write_line(BIT(data, 2));
+	m_i_folla_m->write_line(BIT(data, 1));
+	m_i_folla_b->write_line(BIT(data, 0));
```

Comment:
This has lost the English translations from the comments.

Reply:
Done